lib/phabricator/phab_object.rb

```diff
@@ -2,18 +2,126 @@
 
 module Phabricator
   class PhabObject
-    attr_reader :phid
+    def self.inherited(other)
```

Comment:
This seems like we could use a Gem for providing this, but I honestly don't know if there is one. Just seems like there _should_ be on that exists.